compiler/src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -713,7 +725,41 @@ object Parsers {
           case _                      =>
             syntaxErrorOrIncomplete(IllegalLiteral())
             null
-        })
+        }
+        Literal(Constant(value))
+      }
+
+      atSpan(negOffset) {
+        if (in.token == QUOTEID) {
+          if ((staged & StageKind.Spliced) != 0 && isIdentifierStart(in.name(0))) {
+            val t = atSpan(in.offset + 1) {
+              val tok = in.toToken(in.name)
+              tok match {
+                case TRUE | FALSE | NULL => literalOf(tok)
+                case THIS => This(EmptyTypeIdent)
+                case _ => Ident(in.name)
+              }
+            }
+            in.nextToken()
+            Quote(t)
+          }
+          else {
+            migrationWarningOrError(em"""symbol literal '${in.name} is no longer supported,
```

Comment:
We currently only support `'x`, `'true`, `'false` and `'null` within a `'{ ... }` or `${ ... }`. Is there any reason not to support them outside when `isScala2Mode == false`?

Reply:
We could also consider reintroducing  the syntax `'()` for quoted unit. It look more natural than `'{}`